hyperspy/io_plugins/unbcf_fast.pyx

```diff
@@ -0,0 +1,699 @@
+import cython
+import numpy as np
+import sys
+
+cdef int byte_order
+if sys.byteorder == 'little':
+    byte_order = 0
+else:
+    byte_order = 1
+
+from libc.stdint cimport uint8_t, uint16_t, uint32_t, uint64_t, int64_t
+from libc.string cimport memcpy, memmove
+
+# structs for headers:
+
+cdef struct Map_Header: #size 8
+    uint32_t height
+    uint32_t width
+
+cdef packed struct Line_Header: #size 4
+    uint32_t pixels
+
+cdef packed struct Additional_pulses: #size 4
+    uint32_t size
+
+cdef packed struct Pixel_Header: #size 22
+    uint32_t pixel_x
+    uint16_t chan1
+    uint16_t chan2
+    uint32_t dummy         # unknown purpoise
+    uint16_t flag
+    uint16_t data_size1
+    uint16_t n_of_pulses
+    uint16_t data_size2
+    uint16_t dummy2        # unknown purpoise
+
+# instructivelly packed array structs:
+
+cdef packed struct Bunch_head: #size 2bytes
+    uint8_t size
+    uint8_t channels
+
+cdef struct Gain8:
+    uint8_t val
+
+cdef struct Gain16:
+    uint16_t val
+
+cdef struct Gain32:
+    uint32_t val
+
+cdef struct Gain64:
+    uint64_t val
+
+cdef struct Val16:
+    uint16_t val
+
+cdef struct Val32:
+    uint32_t val
```

Comment:
I will try and see if that will work... I noticed some bugs in some not essential code in python code (night work...) and just updated... 
I used structs as I find it much easier  and cleaner to cast pointer to struct than to ctypedef'ed variables... those still will need to be be accesed by Val32[0] (probably, I just got rid of those stucts and '.val', and it do not compile anymore) notation instead of Val32.val, and casting getts complicated... I am very inexperienced with c and cython, and the syntax of python is quite mixing up...  I will see how this will go